std/concurrency.d

```diff
@@ -2268,6 +2273,26 @@ private
 
 version (unittest)
 {
+    void changeScheduler(Scheduler s)
+    {
+        import core.memory : GC;
+        import core.thread : Thread;
+
+        bool sleepFirst = false;
+        while (Thread.getAll().length != 1)
+        {
+            // wait for all other threads to terminate, use GC.collect to help progress threads by
+            // running finalizers (which might trigger OwnerTerminated or LinkTerminated that other
+            // threads are waiting for)
+            if (sleepFirst)
+                Thread.sleep(dur!("msecs")( 10 ));
+            else
+                sleepFirst = true;
+            GC.collect;
```

Comment:
That looks frightening? Do we really only send Owner/LinkTerminated messages when the thread object get's collected? Sounds horribly unreliable as the other peer might hold some (implicit) reference to the thread.
If so we should add some onThreadExit hook to core.thread or wrap the thread function with some `scope (exit)` guard.